metapi/metapi.js

```diff
@@ -402,5 +456,19 @@ window.metapi = {};
     sock = sockets[key];
 
     sock.addCallback(messageCallback);
+
+    if (closeCallback) {
+      sock.addCloseCallback(closeCallback);
+    }
+  };
+
+  /**
+   * Registers sockets / listeners for socckets that were requested while an dependency was still loading.
+   */
+  metapi.watchPendingSockets = function () {
```

Comment:
This feels like something the library should handle, instead of making userscripts call it